### PR TITLE
Normal generator

### PR DESCRIPTION
**Pull request by mchernov-intel**
moved from #1425

### CODE REVIEW

File: cpp/oneapi/dal/algo/pca/test/batch.cpp [hunk 1]
```diff
@@ -211,6 +214,28 @@ TEMPLATE_LIST_TEST_M(pca_batch_test, "pca common flow", "[pca][integration][batc
...
+    const auto data_table_id = this->get_homogen_table_id();
+
+    const std::int64_t component_count = GENERATE_COPY(0,
+                                                       1,
+                                                       data.get_column_count(),
+                                                       data.get_column_count() - 1,
+                                                       data.get_column_count() / 2);
+
+    this->general_checks(data, component_count, data_table_id);
+}
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/pca/test/batch.cpp:237]: I don't think we need normal distribution tests for PCA. Please consult with @vlad-nazarov, he needed normal distribution for kernel functions testing

File: cpp/oneapi/dal/algo/pca/test/batch.cpp [hunk 2]
```diff
@@ -163,20 +163,20 @@ class pca_batch_test : public te::algo_fixture {
         const auto V = la::matrix<double>::wrap(eigenvectors);
         const auto E = la::matrix<double>::eye(V.get_row_count());
         const auto VxVT = la::dot(V, V.t());
-        const double diff = la::l_inf_norm(VxVT, E);
+        const double diff = la::abs_error(VxVT, E);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/pca/test/batch.cpp:166]: What is the tolerance? Please specify it explicitly

File: cpp/oneapi/dal/test/engine/dataframe_actions.hpp [hunk 3]
```diff
@@ -94,6 +94,47 @@ class dataframe_builder_action_fill_uniform : public dataframe_builder_action {
     std::int64_t seed_ = 7777;
 };
 
+class dataframe_builder_action_fill_normal : public dataframe_builder_action {
+public:
+    explicit dataframe_builder_action_fill_normal(double a, double b, std::int64_t seed)
```
Comment (rlnx) [inline cpp/oneapi/dal/test/engine/dataframe_actions.hpp:99]: I repeat the same question again: what is `a` and `b` for normal distribution? Use more appropriate naming. For example: `loc` and `std`.

File: cpp/oneapi/dal/algo/pca/test/batch.cpp [hunk 4]
```diff
@@ -211,6 +214,28 @@ TEMPLATE_LIST_TEST_M(pca_batch_test, "pca common flow", "[pca][integration][batc
...
 
+TEMPLATE_LIST_TEST_M(pca_batch_test,
+                     "pca common flow normal distribution",
+                     "[pca][integration][batch]",
+                     pca_types) {
+    SKIP_IF(this->not_available_on_device());
+
+    const te::dataframe data =
+        GENERATE_DATAFRAME(te::dataframe_builder{ 100, 10 }.fill_normal(5.0, 2.0),
+                           te::dataframe_builder{ 100000, 10 }.fill_normal(5.0, 1.5));
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/pca/test/batch.cpp:225]: Does all features have common distribution?

**Review by KulikovNikita — APPROVED** [review]

**Review by vlad-nazarov — APPROVED** [review]

File: cpp/oneapi/dal/algo/linear_kernel/test/batch.cpp [hunk 5]
```diff
@@ -86,8 +86,13 @@ class linear_kernel_batch_test : public te::algo_fixture {
                              const table& y_data,
                              const table& result_values) {
         auto reference = compute_reference(scale, shift, x_data, y_data);
+        const table reference_table = dal::detail::homogen_table_builder{}
+                                          .reset(reference.get_array(),
+                                                 reference.get_row_count(),
+                                                 reference.get_column_count())
+                                          .build();
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/linear_kernel/test/batch.cpp:93]: Why reference is converted to table?

File: cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp [hunk 6]
```diff
@@ -83,7 +83,12 @@ class rbf_kernel_batch_test : public te::algo_fixture {
                              const table& result_values) {
         auto reference = compute_reference(sigma, x_data, y_data);
         const double tol = te::get_tolerance<Float>(1e-4, 1e-9);
-        const double diff = la::l_inf_norm(reference, la::matrix<double>::wrap(result_values));
+        const table reference_table = dal::detail::homogen_table_builder{}
+                                          .reset(reference.get_array(),
+                                                 reference.get_row_count(),
+                                                 reference.get_column_count())
+                                          .build();
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp:90]: Why reference is converted to table?

**Review by rlnx — APPROVED** [review]

**Pull request merged**